java/src/test/java/org/hyperledger/fabric/client/ChaincodeEventsTest.java

```diff
@@ -131,11 +155,140 @@ void sends_valid_request_with_specified_start_block_number_using_sign_bit_for_un
+        String transactionId = "TRANSACTION_ID";
+        Checkpointer checkpointer = new InMemoryCheckpointer();
+        checkpointer.checkpointTransaction(blockNumber, Optional.ofNullable(transactionId));
+        EventsRequest<ChaincodeEvent> eventsRequest = network.newChaincodeEventsRequest("CHAINCODE_NAME")
+                .checkpoint(checkpointer)
+                .build();
+
+        try (CloseableIterator<ChaincodeEvent> iter = eventsRequest.getEvents()) {
+            iter.hasNext(); // Interact with iterator before asserting to ensure async request has been made
+        }
+
+        SignedChaincodeEventsRequest signedRequest = mocker.captureChaincodeEvents();
+        ChaincodeEventsRequest request = ChaincodeEventsRequest.parseFrom(signedRequest.getRequest());
+
+        assertChaincodeRequest(request, new ExpectedRequest("NETWORK", "CHAINCODE_NAME", Ab.SeekPosition.TypeCase.SPECIFIED, Optional.ofNullable(checkpointer.getBlockNumber()),checkpointer.getTransactionId()));
+    }
+
+    @Test
+    void sends_valid_request_with_with_start_block_and_checkpointer_chaincode_event() throws  Exception {
+        long blockNumber = 1;
+        ChaincodeEventPackage.ChaincodeEvent event = ChaincodeEventPackage.ChaincodeEvent.newBuilder()
+                .setChaincodeId("CHAINCODE_NAME")
+                .setTxId("tx1")
+                .setEventName("event")
+                .setPayload(ByteString.copyFromUtf8("payload1"))
+                .build();
+        ChaincodeEventImpl chaincodeEvent = new ChaincodeEventImpl(blockNumber, event);
+        Checkpointer checkpointer = new InMemoryCheckpointer();
+
+        checkpointer.checkpointChaincodeEvent(chaincodeEvent);
+        EventsRequest<ChaincodeEvent> eventsRequest = network.newChaincodeEventsRequest("CHAINCODE_NAME")
+                .checkpoint(checkpointer)
+                .build();
+
+        try (CloseableIterator<ChaincodeEvent> iter = eventsRequest.getEvents()) {
+            iter.hasNext(); // Interact with iterator before asserting to ensure async request has been made
+        }
+
+        SignedChaincodeEventsRequest signedRequest = mocker.captureChaincodeEvents();
+        ChaincodeEventsRequest request = ChaincodeEventsRequest.parseFrom(signedRequest.getRequest());
+
         assertThat(request.getChannelId()).isEqualTo("NETWORK");
         assertThat(request.getChaincodeId()).isEqualTo("CHAINCODE_NAME");
 
         assertThat(request.getStartPosition().getTypeCase()).isEqualTo(Ab.SeekPosition.TypeCase.SPECIFIED);
-        assertThat(request.getStartPosition().getSpecified().getNumber()).isEqualTo(startBlock);
+        assertThat(request.getStartPosition().getSpecified().getNumber()).isEqualTo(blockNumber);
+        assertThat(request.getAfterTransactionId()).isEqualTo(event.getTxId());
+        assertChaincodeRequest(request, new ExpectedRequest("NETWORK", "CHAINCODE_NAME", Ab.SeekPosition.TypeCase.SPECIFIED, Optional.ofNullable(blockNumber), Optional.of(event.getTxId())));
```

Comment:
Lots of duplication here. Needs tidying up and would suggest referencing `chaincodeEvent.getBlockNumber()` and `chaincodeEvent.getTransactionId()` as expected values